src/peft/tuners/lora/config.py

```diff
@@ -121,3 +122,10 @@ def __post_init__(self):
         self.target_modules = (
             set(self.target_modules) if isinstance(self.target_modules, list) else self.target_modules
         )
+        # if target_modules is a regex expression, then layers_to_transform should be None
+        if isinstance(self.target_modules, str) and self.layers_to_transform is not None:
+            raise ValueError("`layers_to_transform` cannot be used when `target_modules` is a regex expression.")
```

Comment:
```suggestion
            raise ValueError("`layers_to_transform` cannot be used when `target_modules` is str.")
```

I think this is better, because the user might not be aware that the str they are passing is used as a regex.